cloudflare/resource_cloudflare_api_token.go

```diff
@@ -0,0 +1,248 @@
+				Computed:  true,
+				Sensitive: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"issued_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"modified_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCloudflareApiTokenCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	name := d.Get("name").(string)
+
+	log.Printf("[INFO] Creating Cloudflare ApiToken: name %s", name)
+
+	t := cloudflare.APIToken{
+		Name:      name,
+		Policies:  resourceDataToApiTokenPolices(d),
+		Condition: resourceDataToApiTokenCondition(d),
+	}
+
+	t, err := client.CreateAPIToken(t)
+	if err != nil {
+		return fmt.Errorf("Error creating Cloudflare ApiToken %q: %s", name, err)
+	}
+
+	d.SetId(t.ID)
+	d.Set("status", t.Status)
+	d.Set("issued_on", t.IssuedOn.Format(time.RFC3339Nano))
+	d.Set("modified_on", t.ModifiedOn.Format(time.RFC3339Nano))
+	d.Set("value", t.Value)
+
+	return nil
+}
+
+func resourceDataToApiTokenCondition(d *schema.ResourceData) *cloudflare.APITokenCondition {
+	ipIn := []string{}
+	ipNotIn := []string{}
+	if ips, ok := d.GetOk("request_ip_in"); ok {
+		ipIn = expandInterfaceToStringList(ips)
+	}
+	if ips, ok := d.GetOk("request_ip_not_in"); ok {
+		ipNotIn = expandInterfaceToStringList(ips)
+	}
+
+	return &cloudflare.APITokenCondition{
+		RequestIP: &cloudflare.APITokenRequestIPCondition{
+			In:    ipIn,
+			NotIn: ipNotIn,
+		},
+	}
+}
+
+func resourceDataToApiTokenPolices(d *schema.ResourceData) []cloudflare.APITokenPolicies {
+	policies := d.Get("policy").([]interface{})
+	var cfPolicies []cloudflare.APITokenPolicies
+
+	for _, p := range policies {
+		policy := p.(map[string]interface{})
+
+		permissionGroups := expandInterfaceToStringList(policy["permission_groups"])
+		var cfPermissionGroups []cloudflare.APITokenPermissionGroups
+		for _, pg := range permissionGroups {
+			cfPermissionGroups = append(cfPermissionGroups, cloudflare.APITokenPermissionGroups{
+				ID: pg,
+			})
+		}
+
+		cfResources := map[string]interface{}{}
+		for k, v := range policy["resources"].(map[string]interface{}) {
+			// value can be object or just a string ("*"), try to convert it to map
+			obj := map[string]string{}
+			if err := json.Unmarshal([]byte(v.(string)), &obj); err == nil {
+				cfResources[k] = obj
+			} else {
+				cfResources[k] = v
+			}
+		}
+
+		cfPolicies = append(cfPolicies, cloudflare.APITokenPolicies{
+			Effect:           policy["effect"].(string),
+			Resources:        cfResources,
+			PermissionGroups: cfPermissionGroups,
+		})
+	}
+
+	return cfPolicies
+}
+
+func resourceCloudflareApiTokenRead(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*cloudflare.API)
+	tokenID := d.Id()
+
+	t, err := client.GetAPIToken(tokenID)
+
+	log.Printf("[DEBUG] Cloudflare APIToken: %+v", t)
+	log.Printf("[DEBUG] Cloudflare APIToken error: %#v", err)
```

Comment:
I don't think we need this as have error handling and reporting at https://github.com/cloudflare/terraform-provider-cloudflare/pull/862/files#diff-9267bd9250e36c7a5f656a1602a7847b6e144773edaa39c245bac468dbe57f93R183